File: source/includes/fts-tutorial/synonyms/explicit-advanced-query.go
package main
import (
	"context"
	"fmt"
	"time"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)
// define structure of movies collection
type MovieCollection struct {
	title   string    `bson:"Title,omitempty"`
}
func main() {
	var err error
    // connect to the Atlas cluster
	ctx := context.Background()
	client, err := mongo.Connect(ctx, options.Client().ApplyURI("<connection-string>"))
	if err != nil {
		panic(err)
	}
	defer client.Disconnect(ctx)
	// set namespace
	collection := client.Database("sample_mflix").Collection("movies")
    // define pipeline
    searchStage := bson.D{{"$search", bson.M{
		"compound": bson.M{
      "must": bson.A{
        bson.M{
          "text": bson.D{
            {"path", "title"},{"query", "boat"},{"synonyms", "transportSynonyms"},
          },
        },
        bson.M{
          "text": bson.D{
            {"path", "title"},{"query", "hat"},{"synonyms", "attireSynonyms"},
          },
        },
      },
		},
    }}}
	limitStage := bson.D{{"$limit", 10}}
	projectStage := bson.D{{"$project", bson.D{{"title", 1}, {"_id", 0}, {"score", bson.D{{"$meta", "searchScore"}}}}}}
	// specify the amount of time the operation can run on the server
    opts := options.Aggregate().SetMaxTime(5 * time.Second)
	// run pipeline
	cursor, err := collection.Aggregate(ctx, mongo.Pipeline{searchStage, limitStage, projectStage}, opts)
	if err != nil {
	  panic(err)
    }
    // print results
	var results []bson.D
	if err = cursor.All(context.TODO(), &results); err != nil {
		panic(err)
	}
	for _, result := range results {
		fmt.Println(result)
	}
}

File: content/search/source/includes/synonyms-tutorial/code-snippets/csharp/ExplicitAdvancedQuery.cs
using MongoDB.Bson;
using MongoDB.Driver;
using System;
using System.Threading.Tasks;

class Program
{
    public static async Task<int> Main(string[] args)
    {
        try
        {
            // Connection string to your MongoDB deployment
            var connectionString = "<connection-string>";

            // Create a MongoDB client
            var client = new MongoClient(connectionString);

            // Get a handle on the database and collection
            var database = client.GetDatabase("sample_mflix");
            var collection = database.GetCollection<BsonDocument>("movies");

            // Create the first text search clause for boat
            var text1 = new BsonDocument
            {
                { "text", new BsonDocument
                    {
                        { "path", "title" },
                        { "query", "boat" },
                        { "synonyms", "transportSynonyms" }
                    }
                }
            };

            // Create the second text search clause for hat
            var text2 = new BsonDocument
            {
                { "text", new BsonDocument
                    {
                        { "path", "title" },
                        { "query", "hat" },
                        { "synonyms", "attireSynonyms" }
                    }
                }
            };

            // Create the search stage with compound operator using should array
            var shouldArray = new BsonArray { text1, text2 };

            // Create the search stage
            var searchStage = new BsonDocument
            {
                { "$search", new BsonDocument
                    {
                        { "index", "default" },
                        { "compound", new BsonDocument
                            {
                                { "should", shouldArray }
                            }
                        }
                    }
                }
            };

            // Create the limit stage
            var limitStage = new BsonDocument
            {
                { "$limit", 10 }
            };

            // Create the project stage
            var projectStage = new BsonDocument
            {
                { "$project", new BsonDocument
                    {
                        { "title", 1 },
                        { "_id", 0 },
                        { "score", new BsonDocument
                            {
                                { "$meta", "searchScore" }
                            }
                        }
                    }
                }
            };

            // Create the aggregation pipeline
            var pipeline = new[] { searchStage, limitStage, projectStage };

            // Set options (max time 5 seconds = 5000 ms)
            var options = new AggregateOptions
            {
                MaxTime = TimeSpan.FromMilliseconds(5000)
            };

            // Execute the aggregation
            using (var cursor = await collection.AggregateAsync<BsonDocument>(pipeline, options))
            {
                // Process and display the results
                await cursor.ForEachAsync(doc =>
                {
                    Console.WriteLine(doc.ToJson());
                });
            }

            return 0;
        }
        catch (Exception e)
        {
            Console.Error.WriteLine($"Error: {e.Message}");
            return 1;
        }
    }
}